packages/shared/src/hooks/useChat.ts

```diff
@@ -176,13 +81,96 @@ export const useChat = ({ id: idFromProps, query }: UseChatProps): UseChat => {
         sourceRef.current.close();
       }
 
-      setSearchQuery(undefined);
+      setSessionId(undefined);
+
+      let queryKey: ReturnType<typeof generateQueryKey> = null;
+
+      const setSearchQuery = (chunk: Partial<SearchChunk>) => {
+        client.setQueryData<Search>(queryKey, (previous) =>
+          updateSearchData(previous, chunk),
+        );
+      };
+
+      const onMessage = (event: MessageEvent) => {
+        try {
+          const data: UseChatMessage = JSON.parse(event.data);
+
+          switch (data.type) {
+            case UseChatMessageType.SessionCreated: {
+              const payload = data.payload as CreatePayload;
+              queryKey = generateQueryKey(RequestKey.Search, user, payload.id);
+              setSessionId(payload.id);
+
+              client.setQueryData(
+                queryKey,
+                initializeSearchSession({
+                  ...payload,
+                  createdAt: new Date(),
+                  status: data.status,
+                  prompt: value,
+                }),
+              );
+
+              break;
+            }
+            case UseChatMessageType.WebSearchFinished:
+              setSearchQuery({
+                sources: (data.payload as SourcesMessage).sources,
+                status: data.status,
+              });
+              break;
+            case UseChatMessageType.WebResultsFiltered:
+              setSearchQuery({ status: data.status });
+              break;
+            case UseChatMessageType.StatusUpdated:
+              setSearchQuery({ status: data.status });
+              break;
+            case UseChatMessageType.NewTokenReceived:
+              setSearchQuery({
+                response: (data.payload as TokenPayload).token,
+              });
+              break;
+            case UseChatMessageType.Completed: {
+              setSearchQuery({ completedAt: new Date() });
+              sourceRef.current?.close();
+              break;
+            }
+            case UseChatMessageType.Error:
+              setSearchQuery({ error: data.payload as SearchChunkError });
+              sourceRef.current?.close();
+              break;
+            case UseChatMessageType.SessionFound: {
+              const sessionData = data.payload as Search;
+              client.setQueryData(queryKey, sessionData);
+              sourceRef.current?.close();
+              break;
+            }
+            default:
+              break;
+          }
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error('[EventSource][message] error', error);
```

Comment:
Not sure why we even need this if it's just a console log, is it handled somewhere actually?